surface/include/pcl/surface/impl/concave_hull.hpp

```diff
@@ -182,416 +182,412 @@ pcl::ConcaveHull<PointInT>::performReconstruction (PointCloud &alpha_shape, std:
+  std::vector<double> points;
 
   for (size_t i = 0; i < cloud_transformed.points.size (); ++i)
   {
-    points[i * dim_ + 0] = static_cast<coordT> (cloud_transformed.points[i].x);
-    points[i * dim_ + 1] = static_cast<coordT> (cloud_transformed.points[i].y);
+    points.push_back(cloud_transformed.points[i].x);
+    points.push_back(cloud_transformed.points[i].y);
 
     if (dim_ > 2)
-      points[i * dim_ + 2] = static_cast<coordT> (cloud_transformed.points[i].z);
+      points.push_back(cloud_transformed.points[i].z);
   }
 
   // Compute concave hull
-  exitcode = qh_new_qhull (dim_, static_cast<int> (cloud_transformed.points.size ()), points, ismalloc, flags, outfile, errfile);
-
-  if (exitcode != 0)
-  {
-    PCL_ERROR ("[pcl::%s::performReconstrution] ERROR: qhull was unable to compute a concave hull for the given point cloud (%lu)!\n", getClassName ().c_str (), cloud_transformed.points.size ());
-
-    //check if it fails because of NaN values...
-    if (!cloud_transformed.is_dense)
+  try {
+    orgQhull::Qhull qhull;
+    qhull.runQhull("", dim_, cloud_transformed.size (), points.data(), flags.c_str());
+  
+    if (qhull.qhullStatus() != 0)
     {
-      bool NaNvalues = false;
-      for (size_t i = 0; i < cloud_transformed.size (); ++i)
-      {
-        if (!pcl_isfinite (cloud_transformed.points[i].x) ||
-            !pcl_isfinite (cloud_transformed.points[i].y) ||
-            !pcl_isfinite (cloud_transformed.points[i].z))
-        {
-          NaNvalues = true;
-          break;
-        }
-      }
-
-      if (NaNvalues)
-        PCL_ERROR ("[pcl::%s::performReconstruction] ERROR: point cloud contains NaN values, consider running pcl::PassThrough filter first to remove NaNs!\n", getClassName ().c_str ());
+      throw std::runtime_error(qhull.qhullMessage());
     }
-
-    alpha_shape.points.resize (0);
-    alpha_shape.width = alpha_shape.height = 0;
-    polygons.resize (0);
-
-    qh_freeqhull (!qh_ALL);
-    int curlong, totlong;
-    qh_memfreeshort (&curlong, &totlong);
-
-    return;
-  }
-
-  qh_setvoronoi_all ();
-
-  int num_vertices = qh num_vertices;
-  alpha_shape.points.resize (num_vertices);
-
-  vertexT *vertex;
-  // Max vertex id
-  int max_vertex_id = 0;
-  FORALLvertices
-  {
-    if (vertex->id + 1 > max_vertex_id)
-      max_vertex_id = vertex->id + 1;
-  }
-
-  facetT *facet;    // set by FORALLfacets
-
-  ++max_vertex_id;
-  std::vector<int> qhid_to_pcidx (max_vertex_id);
-
-  int num_facets = qh num_facets;
-  int dd = 0;
-
-  if (dim_ == 3)
-  {
-    setT *triangles_set = qh_settemp (4 * num_facets);
-    if (voronoi_centers_)
-      voronoi_centers_->points.resize (num_facets);
-
-    int non_upper = 0;
-    FORALLfacets
+  
+    qh_setvoronoi_all(qhull.qh());
+  
+    int num_vertices = qhull.vertexCount();
+    alpha_shape.points.resize (num_vertices);
+  
+    // Max vertex id
+    int max_vertex_id = 0;
```

Comment:
Is `Vertex->id()` returning `int`?

Reply:
yes, if you check the qhull headers they use `countT` and that `countT` is defined as an `int`.